src/widget/wtracktableview.cpp

```diff
@@ -1211,6 +1211,13 @@ void WTrackTableView::hideOrRemoveSelectedTracks() {
     restoreCurrentIndex();
 }
 
+void WTrackTableView::editSelectedItem() {
+    if (state() != EditingState) {
+        QKeyEvent keyEvent(QEvent::Type::KeyPress, Qt::Key_F2, Qt::NoModifier);
+        edit(currentIndex(), EditKeyPressed, &keyEvent);
```

Comment:
Why not use the simple method [`edit(index)`](https://doc.qt.io/qt-5/qabstractitemview.html#edit)?

Reply:
Thank you for the suggestion. I replaced the `keyEvent` with `nullptr` in the new (rebased) version of this PR (because I ended up not needing for the future follow-up PR related to `StarEditor`). The `EditKeyPressed` flag on the other hand is specified for two reasons:
* The edit is triggered because of the user pressing the platform edit key (resp. a replacement for that key), so it makes sense to pass that knowledge along using the API provided for this.
* I need the flag in the follow-up PR to distinguish certain user-triggered invocations of `edit()` from those that are triggered internally by `QAbstractItemView`.

Reply:
IIUC you don't need it for this use case. The simple implementation with just the index calls the base function with `AllEditTriggers`
https://github.com/qt/qtbase/blob/e7362764d4931f255d2377462df8ac7a0d4e7c84/src/widgets/itemviews/qabstractitemview.cpp#L1233-L1240
and that starts editing except the trigger is DoubleClicked or SelectedClicked.
https://github.com/qt/qtbase/blob/e7362764d4931f255d2377462df8ac7a0d4e7c84/src/widgets/itemviews/qabstractitemview.cpp#L2749

The EditKey is already enabled here btw, and IIUC that is captured in keyPressEvent and is not relevant for this new slot `editSelectedItem`.
https://github.com/mixxxdj/mixxx/blob/67a41d9dcdb06b37f57be8e88978756140d05ff2/src/widget/wlibrarytableview.cpp#L31

Tl;dr
IMO we don't need the trigger _here_ and I'm curious about your follow-up ; )
Do you mind sharing a link to your WIP branch?

Reply:
Apart from that this PR looks good to me btw.